proposals/lip-0065.md

```diff
@@ -144,23 +158,33 @@ eventSchema = {
                 "dataType": "bytes"
             }
         },
-        "index": {
+        "height": {
             "dataType": "uint32",
             "fieldNumber": 5
+        },
+        "index": {
+            "dataType": "uint32",
+            "fieldNumber": 6
         }
     }
 }
 ```
 
 The maximum size of a serialized event is `MAX_EVENT_SIZE_BYTES`.
 
-#### `moduleID`, `typeID`, and `data`
+#### `module`, `name`, and `data`
+
+The `module` property corresponds to the module to which the event belongs to. Each event has a `name` and JSON schema used to serialize and deserialize the `data` property. The `name` is a string of length at most `MAX_EVENT_NAME_LENGTH`. The value `EVENT_NAME_STANDARD` is reserved for the standard event and cannot be assigned.
 
-The `moduleID` of an event is the ID of the module to which the event belongs to. Each module defines the `typeID` and JSON schema used to serialize and deserialize the `data` property for the events they can emit. The `typeID` is a byte value of length 2. The value `EVENT_TYPE_ID_STANDARD` is reserved for the standard event and cannot be assigned.
+Both `module` and `name` must be alphanumeric strings (lower and upper case characters allowed).
 
 #### `topics`
 
-The event `topics` is an array of unique byte values whose length ranges between `1` and `MAX_TOPICS_PER_EVENT`. By default, it contains one entry for the transaction ID (in the case of events emitted during the transaction processing stages), the constant `EVENT_TOPIC_INIT_GENESIS_STATE` (for events emitted during the 'genesis state initialization' stage), `EVENT_TOPIC_FINALIZE_GENESIS_STATE` (for events emitted during the 'genesis state finalization' stage), `EVENT_TOPIC_BEFORE_TRANSACTIONS` (for events emitted during the 'before transactions execution' stage), or the constant `EVENT_TOPIC_AFTER_TRANSACTIONS` (for events emitted during the 'after transactions execution' stage). An event emitted during transaction processing cannot have any of these reserved constant in the `topics` array.
+The event `topics` is an array of unique byte values whose length ranges between `1` and `MAX_TOPICS_PER_EVENT`. By default, it contains one entry for the transaction ID (in the case of events emitted during the transaction processing stages), the constant `EVENT_TOPIC_INIT_GENESIS_STATE` (for events emitted during the 'genesis state initialization' stage), `EVENT_TOPIC_FINALIZE_GENESIS_STATE` (for events emitted during the 'genesis state finalization' stage), `EVENT_TOPIC_BEFORE_TRANSACTIONS` (for events emitted during the 'before transactions execution' stage), or the constant `EVENT_TOPIC_AFTER_TRANSACTIONS` (for events emitted during the 'after transactions execution' stage). Events emitted as part of the processing of a cross-chain message also contain an entry for the cross-chain message ID. An event emitted during transaction processing cannot have any of these default values in the `topics` array.
```

Comment:
```suggestion
The event `topics` is an array of unique byte values whose length ranges between `1` and `MAX_TOPICS_PER_EVENT`. By default, it contains one entry for the transaction ID (in the case of events emitted during the transaction processing stages), the constant `EVENT_TOPIC_INIT_GENESIS_STATE` (for events emitted during the 'genesis state initialization' stage), `EVENT_TOPIC_FINALIZE_GENESIS_STATE` (for events emitted during the 'genesis state finalization' stage), `EVENT_TOPIC_BEFORE_TRANSACTIONS` (for events emitted during the 'before transactions execution' stage), or the constant `EVENT_TOPIC_AFTER_TRANSACTIONS` (for events emitted during the 'after transactions execution' stage). Events emitted as part of the processing of a cross-chain message also by default always contain an entry for the cross-chain message ID. An event emitted during transaction processing cannot have any of these default values in the `topics` array.
```

Reply:
The topic order is also important and should be specified. I assume that for events emitted in the CCM processing, the transaction ID of the CCU is the first topic and the cross-chain message ID the second topic.